rust/main/utils/run-locally/src/utils.rs

```diff
@@ -119,30 +118,24 @@ pub fn stop_child(child: &mut Child) {
     };
 }
 
-pub fn get_matching_lines(file: &File, search_strings: &[&str]) -> HashMap<String, u32> {
+pub fn get_matching_lines(file: &File, search_strings: &[Vec<String>]) -> Vec<u32> {
```

Comment:
hmm I think if you use `Vec<&str>` you can actually use it as a hashmap key, which would avoid the hardcoding array indices. The indices approach can lead to hard to debug errors later and isn't very easy to reason about at first glance
```suggestion
pub fn get_matching_lines<'a>(
    file: &File,
    search_strings: Vec<Vec<&'a str>>,
) -> HashMap<Vec<&'a str>, u32> {
```

and then in other places like `relayer_restart_invariants_met` you can call as e.g.
```rust
let line_filters = vec![RETRIEVED_MESSAGE_LOG, "CouldNotFetchMetadata"];
...
let matched_logs = get_matching_lines(&relayer_logfile, vec![line_filters.clone()]);

let no_metadata_message_count = matched_logs.get(&line_filters).unwrap().clone();
```

Reply:
ah good idea!